app/models/piece.rb

```diff
@@ -101,4 +121,44 @@ def enemy_at(check_row, check_col)
     enemy = game.piece_at(check_row, check_col)
     enemy if enemy && enemy.is_black != is_black
   end
+
+  def get_path(to_row, to_col)
+    return nil if %w(Rook Bishop Queen).exclude?(type)
+    path = []
+    current_row = row
+    current_col = col
+    until current_row == to_row && current_col == to_col
+      current_row += to_row <=> current_row
+      current_col += to_col <=> current_col
+      path << [current_row, current_col]
+    end
+    path[0..-2]
+  end
+
+  protected
+
+  def valid_move?(to_row, to_col)
+    return false if move_nil?(to_row, to_col)
+    return false if move_out_of_bounds?(to_row, to_col)
+    return false if move_destination_ally?(to_row, to_col)
+    return false unless move_legal?(to_row, to_col)
+    return false if move_obstructed?(to_row, to_col)
+    true
+  end
+
+  def self_check?(to_row, to_col)
+    return false unless move_to!(to_row, to_col)
```

Comment:
Yes, I understood how costly this would be as I was writing it, but I couldn't think of a better way to implement it.